plutus-core/plutus-core/src/PlutusCore/Builtin/Meaning.hs

```diff
@@ -70,57 +70,76 @@ data BuiltinMeaning val cost =
 type HasMeaningIn uni val = (Typeable val, ExMemoryUsage val, HasConstantIn uni val)
 
 -- | A type class for \"each function from a set of built-in functions has a 'BuiltinMeaning'\".
-class (Typeable uni, Typeable fun, Bounded fun, Enum fun, Ix fun, Default (BuiltinVersion fun)) =>
-            ToBuiltinMeaning uni fun where
+class ( Typeable uni
+      , Typeable fun
+      , Bounded fun
+      , Enum fun
+      , Ix fun
+      , Default (BuiltinSemanticsVariant fun)
+      )
+    => ToBuiltinMeaning uni fun
+    where
+
     -- | The @cost@ part of 'BuiltinMeaning'.
     type CostingPart uni fun
 
-    -- | See Note [Versioned builtins]
-    data BuiltinVersion fun
+    -- | See Note [Builtin semantics variants]
+    data BuiltinSemanticsVariant fun
 
     -- | Get the 'BuiltinMeaning' of a built-in function.
-    toBuiltinMeaning :: HasMeaningIn uni val =>
-        BuiltinVersion fun -> fun -> BuiltinMeaning val (CostingPart uni fun)
+    toBuiltinMeaning
+        :: HasMeaningIn uni val
+        => BuiltinSemanticsVariant fun
+        -> fun
+        -> BuiltinMeaning val (CostingPart uni fun)
 
 -- | Get the type of a built-in function.
-typeOfBuiltinFunction :: forall uni fun. ToBuiltinMeaning uni fun =>
-    BuiltinVersion fun -> fun -> Type TyName uni ()
-typeOfBuiltinFunction ver fun =
-    case toBuiltinMeaning @_ @_ @(Term TyName Name uni fun ()) ver fun of
+typeOfBuiltinFunction
+    :: forall uni fun. ToBuiltinMeaning uni fun
+    => BuiltinSemanticsVariant fun
+    -> fun
+    -> Type TyName uni ()
+typeOfBuiltinFunction semvar fun =
+    case toBuiltinMeaning @_ @_ @(Term TyName Name uni fun ()) semvar fun of
         BuiltinMeaning sch _ _ -> typeSchemeToType sch
 
-{- Note [Versioned builtins]
-The purpose of the "versioned builtins" feature is to provide multiple, different denotations
-(implementations) for the same builtin(s).
-An example use of this feature is for "fixing" the behaviour of `ConsByteString` builtin to throw an
-error instead of overflowing its first argument.
+{- Note [Builtin semantics variants] The purpose of the "builtin semantics
```

Comment:
```suggestion
{- Note [Builtin semantics variants] 
The purpose of the "builtin semantics
```

Reply:
Oh, that seems to have broken something.  I'll try to fix it manually.